libs/wallet/src/web3-react/hooks/useSafeAppsSdkInfo.ts

```diff
@@ -9,14 +9,23 @@ export type GnosisSafeSdkInfo = SafeInfo
 export function useSafeAppsSdkInfo(): GnosisSafeSdkInfo | null {
   const [gnosisSafeInfo, setGnosisSafeInfo] = useState<GnosisSafeSdkInfo | null>(null)
   const safeAppsSdk = useSafeAppsSdk()
+  const [intervalTimeout, setIntervalTimeout] = useState<number>(2000)
 
   useEffect(() => {
-    if (!safeAppsSdk) {
-      setGnosisSafeInfo(null)
-    } else {
-      safeAppsSdk.safe.getInfo().then(setGnosisSafeInfo)
-    }
-  }, [safeAppsSdk])
+    const intervalId = setInterval(() => {
+      if (!safeAppsSdk) {
+        setGnosisSafeInfo(null)
+      } else {
+        safeAppsSdk.safe.getInfo().then((safeInfo) => {
+          setGnosisSafeInfo(safeInfo)
+          // if the user is connected, we can check less frequently
+          setIntervalTimeout(safeInfo.isReadOnly ? 10000 : 2000)
+        })
+      }
+    }, intervalTimeout)
+
+    return () => clearInterval(intervalId)
+  }, [safeAppsSdk, intervalTimeout])
```

Comment:
This is a great idea.

One concern I have is that this logic is done in a very general hook. Hooks when used duplicate the logic, meaning that this polling will be done on every instance of the hook. Probably not what you want to do

Look here:
![image](https://github.com/cowprotocol/cowswap/assets/2352112/75c66a15-b2c1-4c68-9e0d-2418d07ac1c5)

Maybe is better to do this once for all the app. I could think `WalletUpdater` is a better place to do it. 

One approach could be for `useSafeAppsSdkInfo()` to return a function to refresh the gnosis safe info on demand. WDYT?